pepper-apis/src/main/resources/org/broadinstitute/ddp/db/dao/JdbiQuestion.sql.stg

```diff
@@ -12,56 +22,117 @@ select qt.question_type_code,
+  and qsc.stable_id in (<questionStableIds>)
+  and rev.end_date is null
+>>
+
+queryQuestionIdByStableIdAndInstanceGuid() ::= <<
+select q.question_id
+  from question as q
+  join question_stable_code as qsc on qsc.question_stable_code_id = q.question_stable_code_id
+  join revision as rev on rev.revision_id = q.revision_id
   join activity_instance as ai on ai.study_activity_id = q.study_activity_id
  where qsc.stable_id = :stableId
    and ai.activity_instance_guid = :instanceGuid
    and rev.start_date \<= ai.created_at
    and (rev.end_date is null or ai.created_at \< rev.end_date)
 >>
 
-queryQuestionDtoByStudyIdStableIdAndUserGuid() ::= <<
-<all_questions_select()>
-JOIN umbrella_study as study on study.umbrella_study_id = qsc.umbrella_study_id
-JOIN study_activity as act on act.study_activity_id = q.study_activity_id and act.study_id = qsc.umbrella_study_id
-JOIN activity_instance as ai on ai.study_activity_id = act.study_activity_id
-JOIN user as u on ai.participant_id = u.user_id
-WHERE
-    qsc.stable_id = :stableId
-    AND u.guid = :userGuid
-    AND study.umbrella_study_id = :studyId
-    AND rev.start_date \<= ai.created_at
-    AND (rev.end_date is null or ai.created_at \< rev.end_date)
->>
+select_all_question_dtos() ::= <<
+select qt.question_type_code as question_type,
+       qsc.stable_id,
+       q.question_id,
+       q.study_activity_id as activity_id,
+       q.question_prompt_template_id as prompt_template_id,
+       q.tooltip_template_id,
+       q.info_header_template_id,
+       q.info_footer_template_id,
+       q.is_restricted,
+       q.is_deprecated,
+       q.is_write_once,
+       q.hide_number,
+       q.revision_id,
+       rev.start_date as revision_start,
+       rev.end_date   as revision_end,
+       coalesce(
+         dq.placeholder_template_id, nq.placeholder_template_id, tq.placeholder_template_id
+       ) as placeholder_template_id,
 
-queryQuestionInfoIfActiveByQuestionId() ::= <<
-<all_questions_select()>
-where q.question_id = :questionId
-  and rev.end_date is null
->>
+       bq.true_template_id,
+       bq.false_template_id,
 
-queryQuestionInfoByQuestionId() ::= <<
-<all_questions_select()>
-where q.question_id = :questionId
->>
+       (select date_render_mode_code from date_render_mode
+         where date_render_mode_id = dq.date_render_mode_id
+       ) as date_render_mode,
+       dq.display_calendar,
+       dqmp.date_question_id as dqmp_id,
+       dqmp.use_month_names,
+       dqyp.date_question_id as dqyp_id,
+       dqyp.years_forward,
+       dqyp.years_back,
+       dqyp.year_anchor,
+       dqyp.first_selected_year,
+       dqyp.allow_future_years,
+       (select group_concat(dft.date_field_type_code order by dqfo.display_order asc separator ',')
+          from date_question_field_order as dqfo
+          join date_field_type as dft on dft.date_field_type_id = dqfo.date_field_type_id
+         where dqfo.date_question_id = dq.question_id
+       ) as date_fields,
```

Comment:
Date questions have at most 3 fields, so it should be safe to do a subquery for it here.